### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by jcabrerizo**
Several dependency versions updated to fix known vulnerabilities:

Changes still in progress as some code needs to be updated

  | Existing version | Updated to
-- | -- | --
org.osgi.core.version | 6.0.0 | 7.0.0
Karaf | 4.2.8 | 4.3.0
WinRM4j | 0.10.0 | 0.11.0
org.apache.cxf:cxf-rt-transports-http | 3.3.5 | 3.3.9
org.eclipse.jetty:jetty-webapp | 9.4.22.v20191022 | 9.4.35.v20201120
com.thoughtworks.xstream:xstream | 1.4.11.1 | 1.4.15
com.fasterxml.jackson. | 2.10.1 | 2.11.3
commons-codec:commons-codec | 1.11 | 1.15
junit:junit | Introduced through:org.apache.cxf:cxf-core@3.3.5 |  
org.apache.httpcomponents:httpclient | 4.5.10 | 4.5.13
org.yaml:snakeyaml | 1.25 | 1.27
org.eclipse.jetty:jetty-server | 9.4.22.v20191022 | 9.4.35.v20201120
com.squareup.okhttp:okhttp | 2.1.2, 2.3.0 *jclouds |  
org.testng:testng | 6.10 | 7.3.0

This PR replaces https://github.com/apache/brooklyn-server/pull/1140

### CODE REVIEW

**Comment by grkvlt** [comment]
Commits 6d4414f and 28ce7db (refer to pax-jetty instead of jetty) seem like they could be merged, otherwise, nice restructuring, lgtm

**Comment by jcabrerizo** [comment]
I dropped the repeated commit @grkvlt thanks for spotting it

File: core/src/main/java/org/apache/brooklyn/util/core/ClassLoaderUtils.java [hunk 1]
```diff
@@ -332,25 +333,44 @@ public URL getResource(String name) {
...
-                .find();
-            if (bundle.isAbsent()) {
+
+            if (Strings.isBlank(version)) {
+                try {
+                    // if the class can be loaded using the default classloader, and it comes from a bundle matching symbolic name, and version not specified,read
+                    // then use that version of the bundle, rather than the latest version
+                    ClassLoader cl = classLoader.loadClass(name).getClassLoader();
+                    if (cl instanceof BundleReference){
+//                    if (cl instanceof BundleClassLoader) {
```
Comment (rdowner) [inline core/src/main/java/org/apache/brooklyn/util/core/ClassLoaderUtils.java:351]: No commented out code please

File: pom.xml [hunk 2]
```diff
@@ -122,7 +122,7 @@
         <activator.servicemix.version>2.9.0</activator.servicemix.version>
         <!-- double-check downstream projects before changing jackson version -->
         <fasterxml.jackson.version>2.11.3</fasterxml.jackson.version>
-        <cxf.version>3.4.1</cxf.version>
+        <cxf.version>3.3.9</cxf.version>
```
Comment (rdowner) [inline pom.xml:127]: Moving *back* a version?

File: karaf/init/src/main/java/org/apache/brooklyn/launcher/osgi/OsgiLauncherImpl.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 package org.apache.brooklyn.launcher.osgi;
 
 import com.google.common.base.Stopwatch;
-import com.sun.xml.bind.v2.TODO;
+//import com.sun.xml.bind.v2.TODO;
```
Comment (rdowner) [inline karaf/init/src/main/java/org/apache/brooklyn/launcher/osgi/OsgiLauncherImpl.java:19]: No commented out code please

File: karaf/features/src/main/feature/feature.xml [hunk 4]
```diff
@@ -111,7 +112,7 @@
         <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.jsr305/${jsr305.bundle.version}</bundle>
         <bundle dependency='true'>mvn:com.google.code.gson/gson/${gson.version}</bundle>
 
-        <bundle dependency='true'>wrap:mvn:javax.activation/activation/${javax-activation.version}</bundle>
+<!--        <bundle dependency='true'>wrap:mvn:javax.activation/activation/${javax-activation.version}</bundle>-->
```
Comment (rdowner) [inline karaf/features/src/main/feature/feature.xml:115]: No commented out code please

File: rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/ScannerInjectHelper.java [hunk 5]
```diff
@@ -15,13 +15,18 @@
...
 import org.apache.cxf.jaxrs.JAXRSServerFactoryBean;
 
 import io.swagger.config.ScannerFactory;
 
 public class ScannerInjectHelper {
     public void setServer(JAXRSServerFactoryBean server) {
-        RestApiResourceScanner.install(server.getResourceClasses());
+        RestApiResourceScanner scanner = new RestApiResourceScanner(server.getResourceClasses());
+        ScannerFactory.setScanner(scanner);
+        // Above method broken in Swagger 1.6.2
```
Comment (rdowner) [inline rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/ScannerInjectHelper.java:29]: What's the consequence of that broken method? Can you expand this comment please?

**Review by rdowner — CHANGES_REQUESTED** [review]
Looks good - just a few questions and minor changes before merging. Also I have not yet run this, but wanted to get the comments to you now.

**Review by grkvlt — CHANGES_REQUESTED** [review]
Some more suggested changes for `ClassLoaderUtils`

**Comment by rdowner** [comment]
@nakomis has created PR #1152 which takes this and adds a few more changes. 1152 is merged, so I'm closing this PR.

**Pull request closed without merging**